vms/administrator/tests/test_report.py

```diff
@@ -54,367 +50,397 @@ def login_admin(self):
 
-        # create job
-        job = ['Developer', '2017-08-21', '2017-08-30', '', created_event]
+        # Create Job
+        job = ['Developer', '2050-05-24', '2050-05-28', '', created_event]
         created_job = create_job_with_details(job)
 
-        # create shift
-        shift = ['2017-08-21', '09:00', '15:00', '10', created_job]
+        # Create Shift
+        shift = ['2050-05-24', '09:00', '15:00', '10', created_job]
         created_shift = create_shift_with_details(shift)
 
-        # shift is assigned to volunteer-one, but hours have not been logged
-        volunteer_shift = register_volunteer_for_shift_utility(
-            created_shift, volunteer)
+        # Shift is assigned to volunteer-one, but hours have not been logged
+        report_page.go_to_admin_report()
+        register_volunteer_for_shift_utility(created_shift, volunteer)
+        report_page.get_page(self.live_server_url, PageUrls.administrator_report_page)
+        # Check admin report with null fields, should not return the above shift
+        # Using multiple tries so to cater to late loading of page.
+        for _ in range(3):
+            try:
+                report_page.fill_report_form(['', '', '', '', ''])
+                break
+            except NoSuchElementException:
+                pass
+        self.assertEqual(report_page.get_alert_box_text(), report_page.no_results_message)
+
+    @staticmethod
+    def register_dataset(parameters):
+        # Register dataset
+        # Register dataset
+        volunteer = create_volunteer_with_details_dynamic_password(parameters['volunteer'])
+        volunteer.organization = parameters['org']
+        volunteer.save()
 
-        report_page = self.report_page
-        # check admin report with null fields, should not return the above shift
-        report_page.fill_report_form(['', '', '', '', ''])
-        self.assertEqual(report_page.get_alert_box_text(),
-                         report_page.no_results_message)
+        # Create Event
+        event = parameters['event']
+        created_event = create_event_with_details(event)
 
+        # Create Job
+        job = parameters['job'] + [created_event]
+        created_job = create_job_with_details(job)
+
+        # Create Shift
+        shift = parameters['shift'] + [created_job]
+        created_shift = create_shift_with_details(shift)
 
-# Failing test case which has been documented - bug #327
-# Test commented out to prevent travis build failure
+        # Create VolunteerShift
+        log_hours_with_details(volunteer, created_shift, parameters['vshift'][0], parameters['vshift'][0])
 
-    def test_check_intersection_of_fields(self):
+    def create_dataset(self):
+        org_one = create_organization_with_details('org-one')
+        org_two = create_organization_with_details('org-two')
+        org_three = create_organization_with_details('org-three')
+        org_four = create_organization_with_details('org-four')
```

Comment:
ghm. not impressive. Please use the concept of Factories here. If not already here, please use http://factoryboy.readthedocs.io/en/latest/orms.html 

Eitherway, please create maybe a function that returns you n number of organizations. This is ugly.

Reply:
>1, yes. Well, if you dont want to include this - atleast wrap this thing up in a factory kind of function and make it return X elements (X being a count param to the function)

Reply:
Just to be clear: you want me to use factory thing in that function or a normal loop to create multiple instance would suffice?

Reply:
it depends on how good you want this code to be. If its not too much work - yeah - factory is what we should have in the codebase. Would be very helpful later as well (specially with fkey relations). 

Else, a fake factory function too. Just look at your timeplan and whatever suits - go for it ?

Reply:
Sure thanks. makes more sense now :+1:

Reply:
Done.